### PR TITLE
single_match_else - single expr/stmt else block corner case

### LINKED ISSUE #3489: FN single_match_else
`clippy 0.0.212 (b2601be 2018-11-27)`

I had this code which did not trigger any lints:
````rust
    let input = match directory {
        Some(value) => value,
        None => {
            return Err((
                ErrorKind::RemoveDirNoArg,
                "No argument assigned to --remove-dir, example: 'git-repos,registry-sources'"
                    .to_string(),
            ))
        }
    };
````

After applying rustfmt, an additional `;` was added:
````rust
    let input = match directory {
        Some(value) => value,
        None => {
            return Err((
                ErrorKind::RemoveDirNoArg,
                "No argument assigned to --remove-dir, example: 'git-repos,registry-sources'"
                    .to_string(),
            ));                                   // <--------- HERE
        }
    };
````

which caused clippy::single_match_else to trigger.

````
warning: you seem to be trying to use match for destructuring a single pattern. Consider using `if let`
   --> src/library.rs:432:17
    |
432 |       let input = match directory {
    |  _________________^
433 | |         Some(value) => value,
434 | |         None => {
435 | |             return Err((
...   |
440 | |         }
441 | |     };
    | |_____^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_match_else
help: try this
    |
432 |     let input = if let Some(value) = directory { value } else {
433 |     return Err((
434 |         ErrorKind::RemoveDirNoArg,
435 |         "No argument assigned to --remove-dir, example: 'git-repos,registry-sources'"
436 |             .to_string(),
437 |     ));
  ...
````

I think the link should also trigger on the unformatted example!

### PR DESCRIPTION
**Pull request by montrivo**
One approach to fix #3489. 
See discussion in the issue.

changelog: single_match_else - single expr/stmt else block corner case fix

### CODE REVIEW

File: tests/ui/single_match_else.rs [hunk 1]
```diff
@@ -32,4 +34,50 @@ macro_rules! unwrap_addr {
...
+    match Some(1) {
+        Some(a) => println!("${:?}", a),
+        None => return,
+    }
+
+    // don't lint here
+    match Some(1) {
+        Some(a) => println!("${:?}", a),
+        None => return,
+    }
```
Comment (matthiaskrgr) [inline tests/ui/single_match_else.rs:52]: is this supposed to be the same?

Comment (montrivo) [reply tests/ui/single_match_else.rs:52]: ah my bad, this is supposed to be the `{return}` case

Comment (montrivo) [reply tests/ui/single_match_else.rs]: I know now, forgot to skip rustfmt on those cases.

**Comment by matthiaskrgr** [comment]
Thanks,  can you please run rustfmt? I see some trailing whitespaces.

**Comment by montrivo** [comment]
> Thanks, can you please run rustfmt? I see some trailing whitespaces.
`cargo dev fmt` did modify anything. I spotted two trailing spaces on comments which I removed.

**Comment by matthiaskrgr** [comment]
>cargo dev fmt did modify anything. 

That's strange.
In theory `cargo fmt --all` should also work in most cases (but it will skip tests I think).

Thanks!

@bors r+
